psi4/src/psi4/libmints/sointegral.cc

```diff
@@ -149,6 +149,7 @@ void OneBodySOInt::compute(std::vector<SharedMatrix> results) {
                     const SOTransformShell &s2 = t2.aoshell[j];
 
                     ob_->compute_shell(s1.aoshell, s2.aoshell);
+                    const auto &aobuf = ob_->buffers();
```

Comment:
not in this case.  the `compute()` that takes a single result matrix always uses buffer 0.  the version that takes a vector of matrices is for things like multipoles where there are different chunks of integrals corresponding to each operator (e.g. mu_x, mu_y, mu_z) and there we want to deference each chunk individually, rather than always read the first one.